unsafe/src/main/java/org/apache/spark/unsafe/bitset/BitSetMethods.java

```diff
@@ -71,7 +72,13 @@ public static boolean isSet(Object baseObject, long baseOffset, int index) {
    * Returns {@code true} if any bit is set.
    */
   public static boolean anySet(Object baseObject, long baseOffset, long bitSetWidthInBytes) {
-    for (int i = 0; i <= bitSetWidthInBytes; i++) {
+    int widthInLong = (int)(bitSetWidthInBytes / SIZE_OF_LONG);
+    for (int i = 0; i <= widthInLong; i++) {
+      if (PlatformDependent.UNSAFE.getLong(baseObject, baseOffset + i) != 0) {
+        return true;
+      }
+    }
+    for (int i = (int)(SIZE_OF_LONG * widthInLong); i < bitSetWidthInBytes; i++) {
```

Comment:
Ah, sorry for overlooking that.  I guess I was thinking of optimizing for the case where the bitset width was a multiple of the word size.